### PR TITLE
fix: v2 bugs

### PR DESCRIPTION
**Pull request by sethyates**
## Description
Fix a variety of little bugs from V2

## Motivation and Context

Make V2 work.

## How Has This Been Tested?

Local project.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: cmd.go [hunk 1]
```diff
@@ -214,13 +218,24 @@ func (r *Runner) runE(runner interface{}, cfg interface{}) func(cmd *cobra.Comma
...
 		l := log.New()
 
+		loglevelFlag, err := cmd.Flags().GetString("loglevel")
+		if err != nil {
+			return err
+		}
+
 		if module, ok := runner.(fx.Option); ok {
 			if _, ok = cfg.(ProvidesFxOptions); ok {
 				runner = func(ctx context.Context, cfg ProvidesFxOptions) error {
```
Comment (algorithmcardboard) [inline cmd.go:228]: The return value is an error. Would it make sense to store it in a different variable?

Also, I find the usage of module in both line 226 and 239 a little confusing.  Would it make sense to pass the runner in line 239 and ignore the return from type check in line 226?

File: parameter/store.go [hunk 2]
```diff
@@ -42,8 +42,3 @@ type ObjectStore interface {
 	ReadObject(ctx context.Context, p string, out interface{}) error
 	WriteObject(ctx context.Context, p string, in interface{}) error
```
Comment (algorithmcardboard) [inline parameter/store.go:43]: Is the purpose of ReadObject and WriteObject to return values of a particular type?

File: parameter/module.go [hunk 3]
```diff
@@ -22,8 +22,4 @@ package parameter
 
 import "go.uber.org/fx"
 
-var Module = fx.Options(
-	fx.Provide(
-		StoreFromObjectStore,
-	),
-)
+var Module = fx.Options()
```
Comment (algorithmcardboard) [inline parameter/module.go:25]: I am unable to understand why we would want to initialize this here?  The actual parameter implementation needs to come from a different library anyway.  Would it be okay to remove this and the initialization in the orlop/module.go line 14?

**Review by algorithmcardboard — APPROVED** [review]
Have a few comments mainly to understand.  Changes look good otherwise

File: cmd.go (hunk 1 above)
Comment (sethyates) [reply cmd.go:228]: This will eventually get rewritten too. v2.1 :-)

File: parameter/store.go (hunk 2 above)
Comment (sethyates) [reply parameter/store.go:43]: Yes of an object type

**Pull request merged**